hail/python/hail/expr/expressions/typed_expressions.py

```diff
@@ -1623,6 +1623,9 @@ def select(self, *fields, **named_exprs):
             Struct containing specified existing fields and computed fields.
         """
 
+        if len(fields) == 0:
+            return hl.or_missing(hl.is_defined(self), hl.struct(**named_exprs))
```

Comment:
Is this actually an improvement?

I think one easy optimization we should be doing (but aren't currently) in the current IR is to rewrite this:
```
ht.row.select(**ht.row_key, **other_fields)
```
to:
```
ht.row.select(*ht.row_key, **other_fields)
```

Which can be implemented more efficiently (with the right ptype).

Reply:
Yours will generate a Let and an If, to replace a SelectFields.

Reply:
I don't think it's strictly an improvement if we look at the code being generated from the IR, but it does prevent the optimization passes in a Table.show from overflowing the stack. (I think we have more expensive introspection on how InsertFields and SelectFields interacts with bindings, so this at least will help us not do that.)

Reply:
I really think this is a deoptimization.

I think the core problem is that TableShow is implemented in Python and generates HUMONGOUS IRs.

Reply:
can you get the show to run if you, say, quadruple the stack size?

Reply:
(setting java opts)